docs/src/pages/components/rating/BasicRating.tsx

```diff
@@ -20,6 +20,15 @@ export default function BasicRating() {
           setValue(newValue);
         }}
       />
+      <Typography component="legend">Highlight selected value only</Typography>
+      <Rating
+        name="highlight-selected-only"
+        value={value}
+        highlightSelectedOnly
+        onChange={(event, newValue) => {
+          setValue(newValue);
+        }}
+      />
```

Comment:
This breaks the inline preview, which is a strong signal that this shouldn't be demonstrated here. Consider the use case seems niche, I would propose a new `### Radio group` header at the bottom of the page.

I thought it was obvious, we should not use the rating system as it doesn't make sense in this context, instead, I would propose we use the "Custom icon set" as the linked issue demonstrates for the motivation.

Reply:
understood. ill add it under sizes doc.